docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,300 @@
+
+We advise that the *HTTP-JSON API* server and *query store* components to have dedicated
+computation and memory resources available to them. This can be achieved via
+containerization or setting them up on independent physical servers. Ensure that the two
+components are **physically co-located** to reduce network latency for
+communication. The scaling and availability aspects heavily rely on the interactions between
+the core components listed above.
+
+With respect to scaling we recommend to follow the general advice in trying to
+understand the bottlenecks and see if adding additional processing power/memory
+i.e vertical scaling is beneficial.
+
+In general for horizontal scaling purposes , we recommend to treat the
+*HTTP-JSON API* and the *query store database server* as a single unit for scaling.
+While a setup with multiple *HTTP-JSON API* services running against a single backend
+database server is feasible, it maybe a futile exercise if the database server
+itself is the bottleneck.
+
+We can run a redundant setup for the *HTTP-JSON API* by using a reverse proxy server with some
+acceptable routing mechanism and then and having multiple *HTTP-JSON API* servers sit behind it
+each with their own backend database servers and dedicated computation and memory resources.
+
+Users may consider running PostgreSQL backend in a `high availability configuration <https://www.postgresql.org/docs/current/high-availability.html>`__.
+The benefits of this are use-case dependent as this may be more expensive for
+smaller active contract datasets, where re-initializing the cache is cheap and fast.
+
+Finally we recommend using app orchestration utilities which monitor the health of the service
+and perform subsequent operations to ensure availability. These utilities can use the
+`healthcheck endpoints <https://docs.daml.com/json-api/index.html#healthcheck-endpoints>`__
+provided by the *HTTP-JSON API* server. This can also be tied into supporting arbitrary
+autoscaling implementation to ensure minimum number of *HTTP-JSON API* servers on
+failures.
+
+
+Logging
+*******
+
+*HTTP-JSON API* server uses the industry-standard Logback for logging. You can
+read more about that in the `Logback documentation <http://logback.qos.ch/>`__.
+
+The logging infrastructure leverages structured logging as implemented by the
+`Logstash Logback Encoder <https://github.com/logstash/logstash-logback-encoder/blob/logstash-logback-encoder-6.3/README.md>`__.
+
+Logged events should carry information about the request being served by the
+*HTTP-JSON API* server. When using a traditional logging target (e.g. standard output
+or rotating files) this information will be part of the log description.
+Using a logging target compatible with the Logstash Logback Encoder allows to have rich
+logs with structured information about the event being logged.
```

Comment:
Here I would expect some mention of what fields I can expect in the logs, or a link to an existing page describing that. May also be worth explicitly mentioning that the logs are JSON by default? (Assuming they are; I've lost track of that stream of work.)

Reply:
@garyverhaegen-da they are not JSON by default , its plaintext encoding, I'll add a line which states that it is the default. Regarding what fields , I can add a note that commands, endpoints and other request/response details are captured in the logs.

Reply:
Really? I thought we had agreed to make JSON the default across the stack. 🤔